src/components/node.js

```diff
@@ -373,25 +376,41 @@ class Node extends React.Component<INodeProps, INodeState> {
       selected,
     });
 
-    return (
+    const readOnly = this.props.readOnly || data.readOnly === true;
+    const nodeAttributes = readOnly
+      ? {
+          onClick: this.handleDragStart,
+          transform: `translate(${x}, ${y})`,
+          style: { transform: `matrix(1, 0, 0, 1, ${x}, ${y})` },
```

Comment:
This is how uber/react-digraph tackles the problem of rendering a `<g>` element correctly. See https://github.com/uber/react-digraph/blob/master/src/components/node.js, specifically the `render()` function.